### ISSUE TITLE
Auto correction: Removal of redundant parentheses can cause a syntax error

### ISSUE THREAD

**Issue by gimite** [rubocop-hq/rubocop#5516]
See the sample code below. rubocop --auto-correct changes a syntactically correct code into a syntactically incorrect code, which I believe it shouldn't.

--------

## Expected behavior

It should convert into syntactically correct code e.g.:
```ruby
x =
  aaaaaa
    .bbbbbb
    .cccccc
    .dddddd ||
  0
puts(x)
```

Or maybe it should just show warning without making automatic correction.

## Actual behavior

rubocop --auto-correct changes a syntactically correct code into a syntactically incorrect code. See below.

## Steps to reproduce the problem

```
$ cat <<EOF > foo.rb
x =
  (aaaaaa
    .bbbbbb
    .cccccc
    .dddddd
  ) || 0
puts(x)
EOF

$ ruby -c foo.rb
Syntax OK

$ rubocop foo.rb --auto-correct foo.rb
Inspecting 1 file
E

Offenses:

foo.rb:2:3: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a method call.
  (aaaaaa ...
  ^^^^^^^
foo.rb:6:4: E: Lint/Syntax: unexpected token tOROP
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
   || 0
   ^^

1 file inspected, 2 offenses detected, 1 offense corrected

$ cat foo.rb
x =
  aaaaaa
    .bbbbbb
    .cccccc
    .dddddd
   || 0
puts(x)

$ ruby -c foo.rb
foo.rb:6: syntax error, unexpected ||, expecting end-of-input
   || 0
     ^
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.4.1 i686-linux)
```

**Comment by asherkach** [comment]
Is it reasonable for `ParenthesesCorrector` to bail without any change in cases like this? If so, I'll try and put together a fix.

**Comment by drbrain** [comment]
This is still broken:

```
$ rubocop -v
0.68.1
$ rubocop -a t.rb
Inspecting 1 file
E

Offenses:

t.rb:2:3: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a method call.
  (aaaaaa ...
  ^^^^^^^
t.rb:6:4: E: Lint/Syntax: unexpected token tOROP
(Using Ruby 2.2 parser; configure using TargetRubyVersion parameter, under AllCops)
   || 0
   ^^

1 file inspected, 2 offenses detected, 1 offense corrected
$ ruby -v t.rb
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]
t.rb:1: warning: assigned but unused variable - x
t.rb:6: syntax error, unexpected ||, expecting end-of-input
   || 0
```

**Comment by drbrain** [comment]
This is still broken

```
$ cat t.rb
x =
  (aaaaaa
    .bbbbbb
    .cccccc
    .dddddd
  ) || 0
puts(x)

$ ruby -c t.rb
Syntax OK
 
$ rubocop t.rb --auto-correct t.rb
Inspecting 1 file
E

Offenses:

t.rb:1:1: C: [Corrected] Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.
x =
^
t.rb:2:3: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a method call.
  (aaaaaa ...
  ^^^^^^^
t.rb:7:4: E: Lint/Syntax: unexpected token tOROP
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
   || 0
   ^^

1 file inspected, 4 offenses detected, 3 offenses corrected

$ cat t.rb
# frozen_string_literal: true
x =
  aaaaaa
    .bbbbbb
    .cccccc
    .dddddd
   || 0
puts(x)

$ ruby -c t.rb
t.rb:7: syntax error, unexpected ||, expecting end-of-input
   || 0
```
